example/Index.re

```diff
@@ -1,25 +1,25 @@
 open Express;
 
-/* The tests below relies upon the ability to store in the Request 
-   objects abritrary JSON properties. 
+/* The tests below relies upon the ability to store in the Request
+   objects abritrary JSON properties.
 
-   Each middleware will both check that previous middleware 
-   have been called by making properties exists in the Request object and 
-   upon success will themselves adds anothe property to the Request. 
+   Each middleware will both check that previous middleware
+   have been called by making properties exists in the Request object and
+   upon success will themselves adds anothe property to the Request.
 
 */
 
-/* [checkProperty req next property k] makes sure [property] is 
-   present in [req]. If success then [k()] is invoked, otherwise 
+/* [checkProperty req next property k] makes sure [property] is
+   present in [req]. If success then [k()] is invoked, otherwise
    [Next.route] is called with next */
 let checkProperty req next property k => {
-  let reqData = Request.asJsonObject req; 
+  let reqData = Request.asJsonObject req;
   switch (Js.Dict.get reqData property) {
     | None => next Next.route
     | Some x => {
       switch (Js.Json.decodeBoolean x) {
-        | Some b when (b == Js.true_) => k ()
-        | _ => next Next.route
+        | Some b when (b == Js.true_) => k ();
+        | _ => next Next.route;
```

Comment:
I think this is best left to another separate library to decode JSON. I think this library should solely expose the JSON object which would be common to many serialization libraries. For instance if I want to use `ocaml-protoc` to deserialize JSON I would rather not pay the cost of having all those decoding functions.